R/module_teal_data.R

```diff
@@ -38,171 +37,216 @@
+  checkmate::assert_string(id)
+  checkmate::assert_flag(validate_shiny_silent_error)
+  moduleServer(id, function(input, output, session) {
+    output$message <- renderUI({
+      is_shiny_silent_error <- inherits(data(), "shiny.silent.error") && identical(data()$message, "")
+      if (inherits(data(), "qenv.error")) {
         validate(
           need(
             FALSE,
-            paste0(
+            paste(
               "Error when executing the `data` module:",
-              strip_style(paste(data_out$message, collapse = "\n")),
-              "Check your inputs or contact app developer if error persists.",
+              strip_style(paste(data()$message, collapse = "\n")),
+              "\nCheck your inputs or contact app developer if error persists.",
               collapse = "\n"
             )
           )
         )
+      } else if (inherits(data(), "error")) {
+        if (is_shiny_silent_error && !validate_shiny_silent_error) {
+          return(NULL)
+        }
+        validate(
+          need(
+            FALSE,
+            sprintf(
+              "Shiny error when executing the `data` module.\n%s\n%s",
+              data()$message,
+              "Check your inputs or contact app developer if error persists."
+            )
+          )
+        )
       }
+    })
+  })
+}
 
+
+#' @keywords internal
+ui_check_class_teal_data <- function(id) {
+  ns <- NS(id)
+  uiOutput(ns("message"))
+}
+
+#' @keywords internal
+srv_check_class_teal_data <- function(id, data) {
+  checkmate::assert_string(id)
+  moduleServer(id, function(input, output, session) {
+    output$message <- renderUI({
       validate(
         need(
-          checkmate::test_class(data_out, "teal_data"),
-          paste0(
-            "Assertion on return value from the 'data' module failed:",
-            checkmate::test_class(data_out, "teal_data"),
-            "Check your inputs or contact app developer if error persists.",
-            collapse = "\n"
-          )
+          inherits(data(), c("teal_data", "error")),
+          "Did not receive `teal_data` object. Cannot proceed further."
         )
       )
-
-      data_out
     })
+  })
+}
 
-    output$shiny_errors <- renderUI({
-      data_validated()
-      NULL
-    })
+#' @keywords internal
+ui_check_shiny_warnings <- function(id) {
+  ns <- NS(id)
+  uiOutput(NS(id, "message"))
+}
 
-    output$shiny_warnings <- renderUI({
-      if (inherits(data_out_r(), "teal_data")) {
-        is_modules_ok <- check_modules_datanames(modules = modules, datanames = .teal_data_ls(data_validated()))
+#' @keywords internal
+srv_check_shiny_warnings <- function(id, data, modules) {
+  checkmate::assert_string(id)
+  moduleServer(id, function(input, output, session) {
+    output$message <- renderUI({
+      if (inherits(data(), "teal_data")) {
+        is_modules_ok <- check_modules_datanames(modules = modules, datanames = .teal_data_ls(data()))
```

Comment:
Now, after https://github.com/insightsengineering/teal/pull/1344 got merged, `modules$datanames = 'all'`.
Previously it would be `modules$datanames = c('ADTTE', 'ADSL', 'all')` for the case where we had at least 2 transformers. 

For the cases where `datanames` is anything (e.g. `"ADSL"`) the check in `check_modules_datanames` that verifies the discrepancy between those two https://github.com/insightsengineering/teal/blob/af5eb40f515f6be941ab05a014e1da9c43e28165/R/utils.R#L148
does not find inconsistency between `'all'` and `"ADSL"`.

```r
modules <- list(datanames = "all")
datanames <- "ADSL"

(extra_datanames <- setdiff(modules$datanames, c("all", datanames)))
#> character(0)
```

Reply:
I've created an issue for this https://github.com/insightsengineering/teal/issues/1347 
Issue wasn't introduced in this PR and can be addresses anytime (@donyunardi will decide when)

Reply:
Ok, so besides this one issue, the PR is solid rock.